File: Sample/Doudizhu/Server/src/ZyGames.Doudizhu.Bll/Action/Action404.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZyGames.Framework.Common.Log;
using ZyGames.Framework.Game.Service;

namespace ZyGames.Doudizhu.Bll.Action
{
    /// <summary>
    /// 错误日志
    /// </summary>
    public class Action404 : BaseAction
    {
        private string errorInfo = string.Empty;

        public Action404(HttpGet httpGet)
            : base(ActionIDDefine.Cst_Action404, httpGet)
        {
        }

        public override bool TakeAction()
        {
            TraceLog.WriteComplement("客户端崩溃日志记录:{0}",errorInfo);
            return true;
        }

        public override void BuildPacket()
        {
        }

        public override bool GetUrlElement()
        {
            if (httpGet.GetString("errorInfo", ref errorInfo))
            {
                return true;
            }
            return false;
        }
    }
}

File: Sample/GameNotice/release/Script/PyScript/Action/action404.py
"""404错误处理"""
import ReferenceLib
from action import *
from ZyGames.Framework.Common.Log import *

class UrlParam(HttpParam):
    def __init__(self):
        HttpParam.__init__(self)
        self.ErrorInfo = ''


class ActionResult(DataResult):
    def __init__(self):
        DataResult.__init__(self)

def getUrlElement(httpGet, parent):
    urlParam = UrlParam();
    if httpGet.Contains("ErrorInfo"):
        urlParam.ErrorInfo = httpGet.GetStringValue("ErrorInfo")
    else:
        urlParam.Result = False;

    return urlParam;

def takeAction(urlParam, parent):
    actionResult = ActionResult();

    TraceLog.WriteError("Client error:{0}", urlParam.ErrorInfo)
    return actionResult;

def buildPacket(writer, urlParam, actionResult):
    return True;